lib/modding/CModHandler.cpp

```diff
@@ -471,47 +471,93 @@ void CModHandler::afterLoad(bool onlyEssential)
 
 }
 
-void CModHandler::trySetActiveMods(std::vector<TModID> saveActiveMods, const std::map<TModID, CModVersion> & modList)
+void CModHandler::trySetActiveMods(const std::vector<std::pair<TModID, CModInfo::VerificationInfo>> & modList)
 {
-	std::vector<TModID> newActiveMods;
-
-	ModIncompatibility::ModList missingMods;
-
+	auto searchVerificationInfo = [&modList](const TModID & m) -> const CModInfo::VerificationInfo*
+	{
+		for(auto & i : modList)
+			if(i.first == m)
+				return &i.second;
+		return nullptr;
+	};
+	
+	std::vector<TModID> newActiveMods, missingMods, excessiveMods;
+	ModIncompatibility::ModListWithVersion missingModsResult;
+	ModIncompatibility::ModList excessiveModsResult;
+	
 	for(const auto & m : activeMods)
 	{
-		if (vstd::contains(saveActiveMods, m))
+		if(searchVerificationInfo(m))
 			continue;
 
-		auto & modInfo = allMods.at(m);
-		if(modInfo.checkModGameplayAffecting())
-			missingMods.emplace_back(m, modInfo.version.toString());
+		//TODO: support actual disabling of these mods
+		if(getModInfo(m).checkModGameplayAffecting())
+		{
+			excessiveMods.push_back(m);
+			allMods[m].setEnabled(false);

```

Comment:
Correct me if I am wrong but this will mark mod as disabled without actually disabling it.

So
- start game with mod X -> mod marked as enabled
- attempt to start save without mod X -> mod will be marked as disabled, but save will fail to load
- attempt to start save with mod X -> save will fail to load because game thinks that mod is disabled

Actually it seems that such scenario is broken on both 1.3.2 and on your branch - server crash.

Reply:
true, better do not touch mods at all for now